src/main/java/sirius/biz/tycho/academy/OnboardingController.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.tycho.academy;
+
+import sirius.biz.web.BizController;
+import sirius.kernel.commons.Tuple;
+import sirius.kernel.di.std.Part;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.settings.Extension;
+import sirius.web.controller.Routed;
+import sirius.web.http.WebContext;
+import sirius.web.services.InternalService;
+import sirius.web.services.JSONStructuredOutput;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Provides the UI for the onboarding / video academy framework.
+ */
+@Register
+public class OnboardingController extends BizController {
+
+    private static final String PARAM_SEEN_IN_PERCENT = "seenInPercent";
+    private static final String PARAM_STARTED = "started";
+    private static final String PARAM_SKIP = "skip";
+
+    @Part
+    @Nullable
+    private OnboardingEngine onboardingEngine;
+
+    /**
+     * Renders an overview of the available track of the video academy for the given owner.
+     *
+     * @param webContext  the request to respond to
+     * @param target      the target entity
+     * @param accessToken the security token to authenticate the target
+     */
+    @Routed(value = "/academy/:1/:2", priority = 999)
+    public void academy(WebContext webContext, String target, String accessToken) {
+        if (!verifyURISignature(webContext, target, accessToken)) {
+            return;
+        }
+
+        webContext.respondWith()
+                  .template("/templates/biz/tycho/academy/tracks.html.pasta",
+                            onboardingEngine != null ? onboardingEngine.fetchTracks(target) : Collections.emptyList(),
+                            target,
+                            computeURISignature(target),
+                            onboardingEngine.fetchSomeVideo(target).orElse(null));
```

Comment:
This will give a nullpointer. Some lines above is an extra check against onboardingEngine being null so its expected that it can be null?